contracts/base/HooksManager.sol

```diff
@@ -32,8 +31,10 @@ contract HooksManager is OnlyAccountCallable {
      * @param hooks Address of the hooks to be enabled for msg.sender.
      */
     function setHooks(address hooks) external onlyAccount {
-        if (hooks != address(0) && !ISafeProtocolHooks(hooks).supportsInterface(type(ISafeProtocolHooks).interfaceId)) {
-            revert AddressDoesNotImplementHooksInterface(hooks);
+        if (hooks != address(0)) {
+            checkPermittedIntegration(hooks);
+            if (!ISafeProtocolHooks(hooks).supportsInterface(type(ISafeProtocolHooks).interfaceId))
+                revert AccountDoesNotImplementValidInterfaceId(hooks);
```

Comment:
Probably should have flagged before, but why is it "Account" in the revert message when we add a hook?

Reply:
The error is defined here and used multiple times in the code: https://github.com/5afe/safe-core-protocol/blob/ea7f2c3d9b929baa32484c9324a54b833e6e0142/contracts/base/RegistryManager.sol#L13

Other uses:
- While enabling Plugin
- While setting registry
- While setting hooks

We kept only one generic error to optimise contract code size

Perhaps `ContractDoesNotImplementValidInterfaceId` would be more suitable?

Reply:
`ContractDoesNotImplementValidInterfaceId` +1 on that naming :)

Reply:
maybe also log the interface id?